components/ModFacilitySearchbar.vue

```diff
@@ -0,0 +1,178 @@
+            <ul
+                v-if="searchInputValue.trim() !== ''"
+                id="search-list"
+                class="bg-primary-bg shadow-md border-2 border-primary/50 rounded-lg absolute
+                w-full flex flex-col divide-y-2 max-h-64 overflow-y-auto overflow-x-hidden
+                "
+                :class="[
+                    isNearPageBottom ? '-translate-y-full' : '',
+                ]"
+            >
+                <li
+                    v-for="(item, index) in filteredItems"
+                    :id="`search-list-item-${index}`"
+                    :key="item.id"
+                    class="flex justify-between divide-x cursor-pointer"
+                    :class="[
+                        selectedItems.has(item) ? 'bg-primary/90' : '',
+                        selectedItemIndex === index ? 'bg-primary-hover text-primary-inverted' : 'opacity-95',
+                    ]"
+                    @click="handleListItemClick"
+                    @mousedown="handleListItemMouseDown"
+                    @mouseover="() => { handleListItemMouseOver(index) }"
+                >
+                    <div class="flex items-center m-3 gap-3 overflow-x-auto">
+                        <span>{{ index + 1 }}</span>
+                        <div class="flex flex-col overflow-x-auto whitespace-nowrap">
+                            <span class="text-xs">{{ item.id }}</span>
+                            <span>{{ `${item.nameEn} / ${item.nameJa} ` }}</span>
+                        </div>
+                    </div>
+                    <div class="flex items-center">
+                        <SVGCheckMark
+                            class="h-4 m-3"
+                            :class="[
+                                selectedItems.has(item) ? 'opacity-100' : 'opacity-10',
+                            ]"
+                        />
+                    </div>
+                </li>
+                <!-- Fallback for empty search results -->
+                <li
+                    v-if="!filteredItems.length"
+                    class="m-3 cursor-default"
+                >
+                    <span>{{ $t('modFacilitySearchBar.noFacilitiesWereFound') }}</span>
+                </li>
+            </ul>
+        </div>
+    </div>
+</template>
+
+<script lang="ts" setup>
+import { computed, ref } from 'vue'
+import ModSearchBar from './ModSearchBar.vue'
+import SVGCheckMark from '~/assets/icons/check-mark.svg'
+import { useFacilitiesStore } from '#build/imports'
+import type { Facility } from '~/typedefs/gqlTypes'
+
+const selectedItems = defineModel<Set<Facility>>({ required: true })
+
+//The limit for fetching facilities is 400 at the current date (30/11/2024)
+const facilitiesStore = useFacilitiesStore()
+const searchInputElement = ref<HTMLInputElement>()
+const searchInputValue = ref('')
+const filteredItems = ref<Facility[]>([])
+const selectedItemIndex = ref(0)
+
+const handleListScroll = () => {
```

Comment:
nice! looks like you got the component separated! 
I think you can also move the list and the handlers into the component as well. It's good that the events are emitted and we should keep that in case the caller wants to add additional logic on the events. For most of these handlers, it looks like we're updating the selectedItems array which is likely logic we'll want to use for any v-model we pass in. (so we should just move the logic into the component) 

The only one that looks unique to facility is the handleSearchInputChange() which looks appropriate to keep here